pkg/addonfactory/template_agentaddon.go

```diff
@@ -125,9 +131,38 @@ func (a *TemplateAgentAddon) validateTemplateData(file string, data []byte) erro
 	if err != nil {
 		return err
 	}
+
+	// There may be some manifests for hosting cluster only or for managed cluster only, we validate it twice,
+	// one assuming it is Hosted mode and the other assuming it is Default mode.
+	configValues[constants.InstallModeBuiltinValueKey] = constants.InstallModeDefault
+	defaultModeMissingKind := false
 	raw := assets.MustCreateAssetFromTemplate(file, data, configValues).Data
 	_, _, err = a.decoder.Decode(raw, nil, nil)
-	return err
+	if err != nil {
```

Comment:
this seems unnecessary to validate twice. You only use one scheme to do that.

Reply:
Fixed